src/client.rs

```diff
@@ -1280,6 +1280,48 @@ impl<T: Read + Write> Session<T> {
         .and_then(|lines| parse_id_seq(&lines, &mut self.unsolicited_responses_tx))
     }
 
+    /// The [`SETQUOTA` command](https://datatracker.ietf.org/doc/html/rfc2087#section-4.1)
+    ///
+    /// Updates the resource limits for a mailbox. Any previous limits for the named quota
+    /// are discarded.
+    /// Returns the updated quota
+    pub fn set_quota<S: AsRef<str>>(
+        &mut self,
+        quota_root: S,
+        limits: Vec<QuotaLimit>,
+    ) -> Result<Quota> {
+        self.run_command_and_read_response(&format!(
+            "SETQUOTA {} ({})",
+            validate_str("SETQUOTA", "quota_root", quota_root.as_ref())?,
+            limits
+                .iter()
+                .map(|e| e.to_string())
+                .collect::<Vec<_>>()
```

Comment:
Feels unfortunate to have to allocate a `Vec` just to print this. There are a couple of other places I think we might be doing something like this, so maybe let's add a helper like this one from Cargo that can join straight from an iterator:
https://github.com/rust-lang/cargo/blob/576356f414bd663526b24e227a6afefe5ecf554b/src/cargo/util/mod.rs#L73-L97